pkg/retry/etagmismatch_error.go

```diff
@@ -0,0 +1,82 @@
+package retry
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+const (
+	// TODO(mainred): etag mismatch error message is not consistent across Azure APIs, we need to normalize it or add more error messages.
+	// Example error message for CRP
+	// Etag provided in if-match header {0} does not match etag {1} of resource.
+	//   where {0} and {1} are the etags provided in the request and the resource respectively.
+	EtagMismatchPattern = `Etag provided in if-match header ([^\s]+) does not match etag ([^\s]+) of resource`
+
+	EtagMismatchErrorTag = "EtagMismatchError"
+)
+
+type EtagMismatchError struct {
```

Comment:
Do we really need this error? e.g. for LB, we are checking the http status code 	// Invalidate the cache because ETAG precondition mismatch.
	if rerr.HTTPStatusCode == http.StatusPreconditionFailed {
		klog.V(3).Infof("LoadBalancer cache for %s is cleanup because of http.StatusPreconditionFailed", ptr.Deref(lb.Name, ""))
		_ = az.lbCache.Delete(*lb.Name)
	}

Reply:
This is not required anymore.
I originally introduced this error because the error of  [the erros of UpdateVMs ](https://github.com/kubernetes-sigs/cloud-provider-azure/blob/793451a4f140baa13c958bb958c9eb9cdeb6738c/pkg/provider/azure_vmss.go#L1467) are aggregated, it's a bit different to tell if etagmismatch error happend. but if we etagmismatch and use errors.As, which internally will iterate the errors wrapped by retry error, we can easily find a etagmismatch in the aggregated error. And that's why I updated the rawerror of retry error [here](https://github.com/kubernetes-sigs/cloud-provider-azure/pull/7698/files#diff-6343963caadf294115d99b1063c041a52225b62d8f1347ed9fc7fd2ad0a90b69R64) as
```
	// RawError indicates the raw error from API.
	// It's beneficial to errors.Is() or errors.As() to check the real error type.
	RawError error
```


We can remove mismatch if we decide to invalidate the cache anyway without checking the exact error.

Reply:
1. can redirect all StatusPreconditionFailed status code to a etagmismatch error?
2. that's how we decide a retry error by [if errors.As(err, &re) ]({https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/vendor/k8s.io/cloud-provider/controllers/service/controller.go#L299)

BTW because our retry error is not api.RetryError, so even if we have retryafter durtion, all errors including retry errors goes to ratelimited retry, we may need to fix this to honor RetryAfter?